core/commands/pin/remotepin.go

```diff
@@ -669,3 +673,16 @@ func getRemotePinServiceInfo(env cmds.Environment, name string) (url, key string
 	}
 	return service.Api.Endpoint, service.Api.Key, nil
 }
+
+func normalizeEndpoint(endpoint string) (string, error) {
+	uri, err := neturl.ParseRequestURI(endpoint)
+	if err != nil || !strings.HasPrefix(uri.Scheme, "http") {
+		return "", fmt.Errorf("service endpoint must be a valid HTTP URL")
+	}
+	// avoid //pins (https://github.com/ipfs/go-ipfs/issues/7826)
+	uri.Path = strings.TrimSuffix(uri.Path, "/")
+	// avoid /pins/pins
+	uri.Path = strings.TrimSuffix(uri.Path, "/pins")
```

Comment:
This could be problematic. In theory this could be the URL the pinning service uses, although that seems like it'd be a terrible idea.

Perhaps we could return an error here like "did you mean X instead of Y"?

Reply:
Added error in 2ffdce5